Question 9 (#5)
eliannyscer/Top-100-Python-Interview: Q9-Python-modules.md

# What are python modules? Name some commonly used built-in modules in Python?

> A Python module is a .py file containing executable code, I mean a set of functions you want to include in your application.

Reviewer: ```suggestion
A Python module is a `.py` file containing executable code or a set of functions you want to include in your application.
```

Some of the commonly used modules are:

* os
* sys
* math
* random
* data time
* JSON

## Additional information

### Create a Module

> To create a module just save the code you want in a file with the file extension .py:

Reviewer: ```suggestion
To create a module just save the code you want in a file with the file extension `.py`:
```

math.pi provides a more precise value for the pi:

```python
import math 
# Print the value of pi 
print (math.pi)

Output:
3.141592653589793
```
  
* `random`: With this module we can `produce anything randomly`.  You can use this module in applications like tic-tac-toe, rock-paper-scissor, etc.

Here a example of a simple program to generate random integers from a range:

```python
import random

# generating a random number from the range 1-100
print(random.randint(1, 100)) 
```

* `data time`: It is `date` and `time` object `combinations`.

```python
import datetime

x = datetime.datetime.now()
print(x) 
```

* `JSON`: It can be used to work with `JSON data`.

```python
import JSON
```

### Runtime and compile runtime

* `Compile-time`: Is the moment where the code you entered is `converted to executable`.
* `Run-time`: Is the moment where `the executable code is running`.

References:

https://www.w3schools.com/python/python_modules.asp
https://www.geeksforgeeks.org/python-sys-module/
https://duckduckgo.com/?t=ffab&q=data+time+module+in+python&ia=web
https://www.w3schools.com/python/python_json.asp

> https://www.geeksforgeeks.org/os-module-python-examples/

Reviewer: Add labels
